src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -60,15 +65,48 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 	private _updateStatus(): void {
 		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
 		if (activeConnection) {
-			this._setConnectionText(activeConnection);
+			let uri = this._connectionManagementService.getConnectionUriFromId(activeConnection.id);
+			let info = this._connectionManagementService.getConnectionInfo(uri);
+			if (this._editorService.activeEditor) {
+				// USE ACTIVE EDITOR INFO AS THE PID WILL BE DIFFERENT FOR EDITOR CONNECTION.
+				let newInfo = this._connectionManagementService.getConnectionInfo(this._editorService.activeEditor.resource.toString());
+				if (newInfo) {
+					info = newInfo;
+				}
+			}
+			if (info && info.pid) {
```

Comment:
You can remove this if/else and just pass in `info.pid` - then just make sure you handle it being undefined in `_setConnectionText`